### ISSUE TITLE
Docs: Add information about binary operation to Add field from calculation

### ISSUE THREAD

**Issue by kmassicotte** [grafana/grafana#29475]
**What would you like to be added**:
Add the ability to organize calculated fields in the Transform panel.

**Why is this needed**:
This would grant users more flexibility in organizing their tables in useful ways, and encourage the use of the Calculated Field transformation.

In order to achieve the column order I desire, I must perform a redundant calculation in a query, then organize in the Transform panel.
Relevant info: Version: v7.2.2 (ad9d408ac2), datasource: Prometheus, transformations applied: Merge, Organize Fields, Add Field From Calculation.

**Comment by ivanahuckova** [comment]
@kmassicotte Not sure if I understand. You can easily, in transformations, calculate fields and right after it apply transformation to organize the table in the way you would like. If I am not understanding correctly, could you please provide more information. 
![image](https://user-images.githubusercontent.com/30407135/100767726-625c7880-33fa-11eb-93af-b336d5b4da22.png)

**Comment by grafanabot** [comment]
Thanks for creating this issue! We think it's missing some basic information. 

Follow the issue template and add additional information that will help us replicate the problem. 
For data visualization issues: 
- Query results from the inspect drawer (data tab & query inspector)
- Panel settings can be extracted in the panel inspect drawer JSON tab

For dashboard related issues: 
- Dashboard JSON can be found in the dashboard settings JSON model view

For authentication and alerting issues, Grafana server logs are useful. 

Happy graphing!

**Comment by kmassicotte** [comment]
Hi @ivanahuckova ,
Sorry, since I hadn't found any existing issues I assumed it was not a bug but a missing feature. It seems to be the former ☺️ .

I've tested a bit more, the bug is reproducible when there is a value in the `Rename XYZ` field in the Organize Fields transformation. When the fields are empty, I'm able to organize the calculated field.

#### Images
With organized fields renamed:
![image](https://user-images.githubusercontent.com/13663054/100780666-fec15380-33d7-11eb-8fd0-54db516a0764.png)

With organized fields blank (default):
![image](https://user-images.githubusercontent.com/13663054/100780781-27e1e400-33d8-11eb-9745-121903e8c196.png)

#### Query results
```
{
  "request": {
    "url": "api/datasources/proxy/24/api/v1/query?query=sum(component%3Acontainer_age_max%7Bservice%3D~%22users%22%7D)%20by%20(colour)&time=1606847066",
    "method": "GET",
    "hideFromInspector": false
  },
  "response": {
    "status": "success",
    "data": {
      "resultType": "vector",
      "result": [
        {
          "metric": {
            "colour": "green"
          },
          "value": [
            1606847066,
            "4303144.350000143"
          ],
          "values": [
            [
              1606847066,
              "4303144.350000143"
            ]
          ]
        },
        {
          "metric": {
            "colour": "blue"
          },
          "value": [
            1606847066,
            "1675250.1199996471"
          ],
          "values": [
            [
              1606847066,
              "1675250.1199996471"
            ]
          ]
        }
      ]
    }
  }
}
```

#### Panel settings
```
{
  "type": "table",
  "title": "Panel Title",
  "gridPos": {
    "x": 0,
    "y": 0,
    "w": 12,
    "h": 8
  },
  "id": 220,
  "targets": [
    {
      "expr": "sum(component:container_age_max{service=~\"$label_service\"}) by (colour)",
      "legendFormat": "",
      "interval": "",
      "refId": "A",
      "format": "table",
      "instant": true
    },
    {
      "expr": "sum(component:container_age_max{service=~\"$label_service\"}) by (colour) + 1",
      "legendFormat": "",
      "interval": "",
      "refId": "B",
      "format": "table",
      "instant": true
    }
  ],
  "options": {
    "showHeader": true
  },
  "datasource": "prometheus",
  "fieldConfig": {
    "defaults": {
      "custom": {
        "align": null,
        "filterable": false
      },
      "thresholds": {
        "mode": "absolute",
        "steps": [
          {
            "value": null,
            "color": "green"
          },
          {
            "value": 80,
            "color": "red"
          }
        ]
      },
      "mappings": []
    },
    "overrides": []
  },
  "pluginVersion": "7.2.2",
  "transformations": [
    {
      "id": "merge",
      "options": {}
    },
    {
      "id": "calculateField",
      "options": {
        "mode": "binary",
        "reduce": {
          "reducer": "sum"
        },
        "binary": {
          "left": "Container Max Age",
          "operator": "-",
          "right": "Container Max Age + 1",
          "reducer": "sum"
        },
        "alias": "Calculated field"
      }
    },
    {
      "id": "organize",
      "options": {
        "excludeByName": {},
        "indexByName": {},
        "renameByName": {
          "Value #A": "Container Max Age",
          "Value #B": "Container Max Age + 1"
        }
      }
    }
  ],
  "timeFrom": null,
  "timeShift": null
}
```

Hope this is sufficient info for you to reproduce on your end and flag as a bug.

**Comment by ivanahuckova** [comment]
Hmmm, so this works as intended. As you can see [here](https://grafana.com/docs/grafana/latest/panels/transformations/#order-of-transformations) transformations are applied one after each other and therefore renaming/creating **A** after you used it in **Add field from calculation** is not going to work. You can add an extra step - `merge => organize (to rename) => add field => organize (to reorder). Something like this: 
![image](https://user-images.githubusercontent.com/30407135/100785252-abb7c280-3410-11eb-8edc-149cac3e66d8.png)

Maybe we could add some kind of warning to **Add fields from calculation** that lets users know that letters that are not the name of one of the fields are not supported and are not going to work.  So it is more clear.

**Comment by kmassicotte** [comment]
I understand now ☺️ .  

The issue I had was due to the order of operations. I had created the Calculated Field (the last transformation in the array) and selected the _renamed_ field in the combo-box, then I re-ordered the transformation to put it ahead of the Organize fields transformation. Since the values stayed the same (i.e. the _renamed field_ offered in the combo-box), I missed the fact that the field hadn't actually been renamed yet due to the order of the transformations.

Resetting the values in the Calculated Field transformation upon moving it up in the order might have clarified the behaviour.

I've resolved the issue on my end now, thank you for clarifying.
